frontend/pages/projects/_id/affective-annotation/index.vue

```diff
@@ -741,9 +750,30 @@ export default {
       }
     },
     async confirm() {
-      await this.$services.example.confirm(this.projectId, this.doc.id)
-      await this.$fetch()
-      this.updateProgress()
+      const canConfirm = this.isAllAffectiveDataAdded()
+      if (canConfirm || this.isProjectAdmin) {
+        this.cannotConfirmAlert = false
+        await this.$services.example.confirm(this.projectId, this.doc.id)
+        await this.$fetch()
+        this.updateProgress()
+      } else {
+        this.cannotConfirmAlert = true
+      }
+    },
+    isAllAffectiveDataAdded() {
+      if (this.project.isSummaryMode) {
+        return this.affectiveSummaryTags.length >= 2 && this.affectiveSummaryImpressions.length >= 2
+      }
+      if (this.project.isEmotionsMode) {
+        return _.keys(this.scaleTypesTextsIds).length === _.keys(this.affectiveScalesValues).length
+      }
+      if (this.project.isOthersMode) {
+        return (
+          _.keys(this.scaleTypesTextsIds).length === _.keys(this.affectiveScalesValues).length &&
```

Comment:
I think rather than checking the keys length, its better to check whether the scales values ids are all present in the scaleTypes or not XD    you can use _.intersectionWith or .every for that. If we are using `length` comparison, then if I added a new scale value in the table (because im a curious admin) this check will always be false.

Reply:
Hello! I agree with your point that checking the length against `scaleTypesTextsIds` is a bad idea because it may contain more scale labels if the admin added their own custom entries. However, I decided to fix it by checking against `affectiveScalesDict`, which is the dict constructed from the static json file inside frontend directory. I think it should work correctly.

I didn't want to go with checking the ids because currently the inputs (sliders) are hard-coded so that if the admin adds additional scale labels, they won't be displayed in annotation page (at least in Emotions and Others modes), and hence checking the ids will cause issues (as there is no way to add responses for these additional scale labels). By checking against the static json file, it should work correctly even if the admin somehow adds additional scale labels, as the most important thing is that the user has provided responses to the dimensions listed in the static json file.

Is it fine? 😅